pyinfra/connectors/util.py

```diff
@@ -179,7 +180,9 @@ def _get_sudo_password(host, use_sudo_password):
             host.connector_data["sudo_password"] = sudo_password
         sudo_password = sudo_password
     elif callable(use_sudo_password):
-        sudo_password = use_sudo_password()
+        with ctx_state.use(state):
```

Comment:
This shouldn't be required as the state context is set for the entire run @ https://github.com/Fizzadar/pyinfra/blob/5fedec46b127d16d59ca73537f90dadff647511d/pyinfra_cli/main.py#L380